### PR TITLE
V2 Refactoring

### PR DESCRIPTION
**Pull request by Chocksy**
This is quite big of a PR. But i thought i would share it.

### Things i did
- i added rubocop gems for development
- i changed the endpoints capabilities to be much more limiting as the V2 API is quite smaller.
- i updated the specs to reflect all of that

### Notes
- the projects endpoints are not fully supported because Dribbble does not have a `GET project/:id` endpoint.

### CODE REVIEW

File: lib/dribbble/project.rb [hunk 1]
```diff
@@ -1,11 +1,20 @@
+# frozen_string_literal: true
+
 require 'dribbble/utils/findable'
+# require 'dribbble/utils/creatable'
+# require 'dribbble/utils/updatable'
+# require 'dribbble/utils/deletable'
 
 module Dribbble
   class Project < Dribbble::Base
+    # TODO: enable these once the GET /projects/:id endpoint gets available!! https://developer.dribbble.com/v2/projects/
```
Comment (Calyhre) [inline lib/dribbble/project.rb:10]: Reading the documentation today, they seem to be available right?

**Review by Calyhre — APPROVED** [review]
That's a very impressive PR, thank you very much for it! I missed the moment you posted it and how Dribbble documentation was at that moment, did they enabled the missing routes on projects since then? Otherwise that looks to me 👏

**Comment by Calyhre** [comment]
Seems like `rubocop-rake` depends on ruby >= 2.4 and that would be why the CI is failing. I think we can safely drop support for rubies < 2.4, what do you think?

File: lib/dribbble/project.rb (hunk 1 above)
Comment (Chocksy) [reply lib/dribbble/project.rb:10]: Hmm. Indeed it seems the endpoints have been added. I'll make the change.

**Comment by Chocksy** [comment]
@Calyhre we should be good now. I added those endpoints back and removed the ruby < 2.4 versions.

**Review by Calyhre — APPROVED** [review]
Looks good to me! Thank you again for all the good stuff 👏

**Comment by Calyhre** [comment]
Published -> https://rubygems.org/gems/dribbble/versions/2.0.0

**Pull request merged**